internal/controller/component_build_controller_service_account.go

```diff
@@ -0,0 +1,523 @@
+
+This PR chnages Service Account used by build pipeline from "appstudio-pipeline" to dedicated to the Component Service Account.
+`
+
+// Need to create a PR that adds to every pipeline definition:
+func (r *ComponentBuildReconciler) setServiceAccountInPipelineDefinition(ctx context.Context, component *appstudiov1alpha1.Component) error {
+	log := ctrllog.FromContext(ctx)
+
+	gitProvider, err := getGitProvider(*component)
+	if err != nil {
+		// There is no point to continue if git provider is not known.
+		return err
+	}
+
+	pacSecret, err := r.lookupPaCSecret(ctx, component, gitProvider)
+	if err != nil {
+		log.Error(err, "error getting git provider credentials secret", l.Action, l.ActionView)
+		// Cannot continue without accessing git provider credentials.
+		return boerrors.NewBuildOpError(boerrors.EPaCSecretNotFound, err)
+	}
+
+	repoUrl := getGitRepoUrl(*component)
+
+	gitClient, err := gitproviderfactory.CreateGitClient(gitproviderfactory.GitClientConfig{
+		PacSecretData:             pacSecret.Data,
+		GitProvider:               gitProvider,
+		RepoUrl:                   repoUrl,
+		IsAppInstallationExpected: true,
+	})
+	if err != nil {
+		return err
+	}
+
+	// getting branch in advance just to test credentials
+	defaultBranch, err := gitClient.GetDefaultBranch(repoUrl)
+	if err != nil {
+		return err
+	}
+	baseBranch := component.Spec.Source.GitSource.Revision
+	if baseBranch == "" {
+		baseBranch = defaultBranch
+	}
+
+	pushPipelinePath := getPipelineRunDefinitionFilePath(component, false)
+	pullPipelinePath := getPipelineRunDefinitionFilePath(component, true)
+
+	pushPipelineDefinitionFound := true
+	var pushPipelineContent []byte
+	pushPipelineContent, err = gitClient.DownloadFileContent(repoUrl, baseBranch, pushPipelinePath)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+		pushPipelineDefinitionFound = false
+	}
+	pullPipelineDefinitionFound := true
+	var pullPipelineContent []byte
+	pullPipelineContent, err = gitClient.DownloadFileContent(repoUrl, baseBranch, pullPipelinePath)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+		pullPipelineDefinitionFound = false
+	}
+	if !pushPipelineDefinitionFound && !pullPipelineDefinitionFound {
+		// No pipeline definitions found
+		log.Info("No pipeline definitions found to migrate")
+		return nil
+	}
+
+	pipelineRunServiceAccountName := getBuildPipelineServiceAccountName(component)
+
+	if pushPipelineDefinitionFound {
+		pushPipelineContent, err = addServiceAccountToPipelineRun(pushPipelineContent, pipelineRunServiceAccountName)
+		if err != nil {
+			log.Error(err, "failed to add Service Account to push pipeline definition")
+			return err
+		}
+	}
+	if pullPipelineDefinitionFound {
+		pullPipelineContent, err = addServiceAccountToPipelineRun(pullPipelineContent, pipelineRunServiceAccountName)
+		if err != nil {
+			log.Error(err, "failed to add Service Account to pull pipeline definition")
+			return err
+		}
+	}
+
```

Comment:
addServiceAccountToPipelineRun should return also boolean if SA was added or not AND
only after that if it needs to be modified create MR